Simplify init of Wrapper (new-style base class)

trosh opened the pull request.

faminours committed:
[misc] [doc] Standardize all commands' doc

Use docstrings for author/brief/descriptions
Use bxiparserconf's new RawDescriptionFilteredHelpFormat when needed.
Set shebang to python2.
Standardize copyrights.

trosh committed:
Simplify init of Wrapper (new-style base class)

trosh committed:
[misc]

trosh committed:
[misc] Pylint on backtrace

jnquintin commented on packaged/bin/bxiconfig:
```diff
@@ -1,13 +1,16 @@
-@copyright 2016  Bull S.A.S.  -  All rights reserved.\n
-           This is not Free or Open Source software.\n
-           Please contact Bull SAS for details about its license.\n
+@author BXIHL <bxihl@atos.net>
```
the authors of the file change?

jnquintin reviewed (commented):
almost ok with it

trosh committed:
[doc] additionnal -> additional

trosh committed:
[misc] Pylint, python2 shebang

trosh committed:
[misc] Clean bxiconfig docstring and __main__ call

trosh commented:
Rebased this PR on top of the one by @faminours, with a minor modif on that one to properly rebase it on origin/develop. The commit by @faminours seems to be non-functional (there's a function with an undefined variable), so this should be fixed before considering either PR.

Pylint work and shebang change is expanded to all commands.

This is all open to feedback; it is however not urgent and might be better off waiting for @faminours to come back into the discussion before going any further.

jnquintin commented:
Why asking for python2 in the main?

trosh commented:
Which main ? The script called as `__main__` ?

jnquintin commented:
For the shabang

trosh commented:
Fixed the minor error @faminours had left in `addepilog`.

About the shebang line: not sure what you mean; I updated the shebang lines because these scripts have not been developped to be run from Python 3 and the shebang line merely controls what interpreter the shell uses for executable scripts in the specific case of scripts being called as executables — it has no effect on modules not executed directly.

Users can (and should) import any module in bxibase from the outside. Therefore, if you wanted to control Python version actively, you would need to add something like `assert (2, 7) <= sys.version_info < (3, 0)` in every Python file. This might also be appropriate by default, until specific files are explicitely checked / fixed / tested for Python 3 compatibility.

About `@author`, I don't really care about the correct course of action so I can revert all corresponding changes. Now I feel like this is a situation where if you don't see a good way to do something, it might be because there's no point doing the thing; removing that content altogether might be better.

jnquintin commented:
Stop thinking that you are the only team working on bxibase. Most of the code comes from personnes now outside of the team bxi. bxibase is used at other place if you want to add your assert you will have a fork. Like the shebang modification is a wrong approche, working on the compatibility would be better. python2.7 death is programmed, a day or another it will come. See the travis tests, they are done in python3. Start coding tests and then you will get emails about the errors. For now the tests in python3 works.
